vimrc

```diff
@@ -106,8 +106,18 @@ inoremap <S-Tab> <c-n>
 " Exclude Javascript files in :Rtags via rails.vim due to warnings when parsing
 let g:Tlist_Ctags_Cmd="ctags --exclude='*.js'"
 
+
 " Index ctags from any project, including those outside Rails
-map <Leader>ct :!ctags -R .<CR>
+function! ReindexCtags()
+  let l:ctags_hook_location = '$(git rev-parse --show-toplevel)/.git/hooks/ctags'
+
+  if exists(l:ctags_hook_location)
+    exec '!'. l:ctags_hook_location
+  else
+    exec "!ctags -R ."
+  endif
+endfunction
+nmap <Leader>ct :call ReindexCtags()<CR>
```

Comment:
Just documenting here in case it's useful to anyone: When I switched to using the `plugin` folder instead of explicitly sourcing vimrc files, I ran into several mysterious instances where a plugin's settings refused to be overridden. I never quite nailed down why that was happening (presumably because I had less control over load order), but the `after` directory proved handy for working around these issues. 

I ended up storing all my general vim settings (leader key bindings, `ftdetect`, `ftplugin` settings, etc) in `~/.vim/after/plugin`, ensuring any intended overrides (e.g., of a plugin's default keybindings) will actually override plugins' default settings. I still use `~/.vim/plugin` for plugin config settings with files named after their plugins, as suggested by @christoomey.

```
:h after-directory

	The defaults for most systems are setup to search five locations:
	1. In your home directory, for your personal preferences.
	2. In a system-wide Vim directory, for preferences from the system
	   administrator.
	3. In $VIMRUNTIME, for files distributed with Vim.
							*after-directory*
	4. In the "after" directory in the system-wide Vim directory.  This is
	   for the system administrator to overrule or add to the distributed
	   defaults (rarely needed)
	5. In the "after" directory in your home directory.  This is for
	   personal preferences to overrule or add to the distributed defaults
	   or system-wide settings (rarely needed).

```